main.js

```diff
@@ -5,21 +5,42 @@ let select = document.querySelector('.categories');
 let textArea = document.querySelector('.pullText');
 
 let buttonClick = document.querySelector('.addButton');
-buttonClick.addEventListener('click', () => {
-    let selectedCategory = select.options[select.selectedIndex].value;
-    let selectedDate = inputDate.value.split('-').splice(1,2).reverse().join('.');
-    let text = textArea.value;
-    let masHelp = [selectedCategory, text, selectedDate];
+let todoItems = [];
 
+function renderTodoItem(item) {
     let tr = document.createElement('tr');
-    for (let i = 0; i < 3; i++) {
+    for (let key in item) {
         let td = document.createElement('td');
-        td.textContent = masHelp[i];
+        td.textContent = key == 'date' ? item[key].slice(5).replace('-', '.') : item[key];
         tr.append(td);
     }
     tr.childNodes[1].className = 'text';
     let table = document.querySelector('.list');
     table.append(tr); 
+}
 
+buttonClick.addEventListener('click', () => {
+    let selectedCategory = select.value;
+    let selectedDate = inputDate.value;
+    let text = textArea.value;
+    let tasks = {
+        category: selectedCategory,
```

Comment:
не обязательно сначала создавать переменные, а потом их пихать в объект.
можно в качестве значений сразу выражения писать
```suggestion
        category: select.value,
```